apps/medon-fe/src/components/SelectTimeSlot/index.tsx

```diff
@@ -1,40 +1,53 @@
 import { timeSlots } from 'utils/constants/options/hourOptions';
+import dayjs from 'dayjs';
 import { useSelectTimeSlot } from './hook';
 import { Container, DrText, SlotActive, TimeSlot, TimeText } from './styles';
 import { SelectTimeSlotProps } from './types';
 
-export default function SelectTimeSlot(
-  props: SelectTimeSlotProps,
-  isActive: string | ''
-) {
-  const { timeSlotsAvailability, selectTime, t } = useSelectTimeSlot(props);
+export default function SelectTimeSlot(props: SelectTimeSlotProps) {
+  const { timeSlotsAvailability, selectTime, t, isActive, selectedDate } =
+    useSelectTimeSlot(props);
 
   return (
     <Container>
-      {timeSlots.map((timeSlot, index) => (
-        <TimeSlot
-          onClick={() => selectTime(timeSlot)}
-          key={timeSlot}
-          style={isActive === timeSlot ? SlotActive : {}}
-          disabled={
-            !Object.values(timeSlotsAvailability).some(
-              (availability) => availability[index]
-            )
-          }
-        >
-          <TimeText>{timeSlot}</TimeText>
-          <DrText>
-            <span>{t('appointment.availableDr')}</span>
-            <span>
-              {
-                Object.values(timeSlotsAvailability).filter(
-                  (availability) => availability[index]
-                ).length
-              }
-            </span>
-          </DrText>
-        </TimeSlot>
-      ))}
+      {timeSlots.map((timeSlot, index) => {
```

Comment:
double check that app will no fail if there no timeslots

Reply:
it does not give errors, it just will not be available day or will not render the slot